File: backend/schemas/mc.py
from pydantic import BaseModel


class McProbabilityLevel(BaseModel):
    label: str
    level: float
    prob_touch: float
    prob_close_above: float


class McResponse(BaseModel):
    symbol: str
    horizon_days: int
    n_paths: int
    sigma: float
    spot: float
    mean_close: float
    std_close: float
    ci_95: tuple[float, float]
    sample_paths: list[list[float]]                # 25 paths × (n_steps + 1)
    bands: dict[str, list[float]]                  # percentile → series
    probabilities: list[McProbabilityLevel]        # one entry per chart annotation level

File: frontend/src/types/mc.ts
import { z } from "zod";

export const McProbabilityLevelSchema = z.object({
  label: z.string(),
  level: z.number(),
  prob_touch: z.number(),
  prob_close_above: z.number(),
});

export const McResponseSchema = z.object({
  symbol: z.string(),
  horizon_days: z.number(),
  n_paths: z.number(),
  sigma: z.number(),
  spot: z.number(),
  mean_close: z.number(),
  std_close: z.number(),
  ci_95: z.tuple([z.number(), z.number()]),
  sample_paths: z.array(z.array(z.number())),
  bands: z.record(z.array(z.number())),
  probabilities: z.array(McProbabilityLevelSchema),
});

export type McResponse = z.infer<typeof McResponseSchema>;
export type McProbabilityLevel = z.infer<typeof McProbabilityLevelSchema>;
